wire/msgtx.go

```diff
@@ -366,6 +367,29 @@ func (msg *MsgTx) TxHash() chainhash.Hash {
 	return chainhash.DoubleHashH(buf.Bytes())
 }
 
+// TxIdFromMsgTxHash generates the transaction ID (txid) of the transaction from
+// the double hash of the transaction.
+func TxIdFromMsgTxHash(txHash chainhash.Hash) string {
```

Comment:
Unnecessary code duplication.

Reply:
I duplicated the code here because I don't think `chainhash.Hash.String` is correct. From my POV, it shouldn't reverse the byte order before creating a string from the hash. From what I understand, that reversal routine should be unique to txid.

In other words, I don't think `chainhash.Hash.String` should be the transaction ID, so I wanted to get away from that method. IMO, duplicating the code in this PR made it clear the relationship between a txhash and a txid. 

However, I only really care about adding `MsgTx.TxID` in this PR. So I've removed this duplicate code.